packages/react/fast-components-react-base/src/data-grid/data-grid-cell.tsx

```diff
@@ -0,0 +1,176 @@
+import React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { isNil } from "lodash-es";
+import { classNames } from "@microsoft/fast-web-utilities";
+import { extractHtmlElement } from "@microsoft/fast-react-utilities";
+import { DataGridCellClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    DataGridCellHandledProps,
+    DataGridCellProps,
+    DataGridCellUnhandledProps,
+} from "./data-grid-cell.props";
+import { DataGridContext, DataGridContextType } from "./data-grid-context";
+
+class DataGridCell extends Foundation<
+    DataGridCellHandledProps,
+    DataGridCellUnhandledProps,
+    {}
+> {
+    public static defaultProps: Partial<DataGridCellProps> = {
+        rowData: null,
+        columnDefinition: null,
+        columnIndex: 0,
+        managedClasses: {},
+    };
+
+    public static displayName: string = "DataGridCell";
+    public static contextType: React.Context<DataGridContextType> = DataGridContext;
+
+    protected handledProps: HandledProps<DataGridCellHandledProps> = {
+        managedClasses: void 0,
+        rowData: void 0,
+        columnDefinition: void 0,
+        columnIndex: void 0,
+    };
+
+    private focusTarget: React.RefObject<any> = React.createRef();
+    private rootElement: React.RefObject<any> = React.createRef();
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactNode {
+        if (
+            isNil(this.context) ||
+            isNil(this.context.dataGridProps) ||
+            isNil(this.props.columnDefinition)
+        ) {
+            return null;
+        }
+
+        const unhandledProps: React.HTMLAttributes<HTMLElement> = {
+            role: "gridcell",
+            tabIndex: this.isFocusCell() ? 0 : -1,
+            onKeyDown: this.handleKeyDown,
+            onFocus: this.handleFocus,
+        };
+
+        if (!isNil(this.props.columnDefinition.cell)) {
+            return this.props.columnDefinition.cell(
+                this.props,
+                this.generateClassNames(),
+                this.props.columnDefinition.columnDataKey,
+                this.rootElement,
+                this.focusTarget,
+                unhandledProps
+            );
+        } else {
+            return this.defaultCellRenderFunction(
+                this.props,
+                this.generateClassNames(),
+                this.props.columnDefinition.columnDataKey,
+                this.rootElement,
+                this.focusTarget,
+                unhandledProps
+            );
+        }
```

Comment:
I would recommend that this is a single argument object like so (without passing through all the props):

```typescript
const config = {
    rowData: this.props.rowData[props.columnDefinition.columnDataKey],
    columnIndex: this.props.columnIndex,
    classNames: this.generateClassNames(),
    columnDataKey: this.props.columnDefinition.columnDataKey,
    rootElement: this.rootElement,
    focusTarget: this.focusTarget,
    unhandledProps: unhandledProps
};
```

you can also cut down on the duplication:

```typescript
return !isNil(this.props.columnDefinition.cell)
    ? this.props.columnDefinition.cell(config)
    : this.defaultCellRenderFunction(config);
```

Reply:
I like the config approach but I think we should keep on passing the whole row data to the cell because someone could create custom cells that render based on multiple data points, for example "display 'paymentAmount' field and highlight if 'dueDate' field is passed.